plugins/lua/stockpiles.lua

```diff
@@ -545,6 +555,7 @@ function StockpilesOverlay:onRenderFrame()
         self.subviews.melt:setOption(config.melt == 1)
         self.subviews.trade:setOption(config.trade == 1)
         self.subviews.dump:setOption(config.dump == 1)
+        self.subviews.forbid:setOption(config.forbid+1)
```

Comment:
this is a bit opaque. I suggest `self.subviews.forbid:setOption(config.forbid)` instead and changing the options in the widget to:
```lua
                        options={{label='Forbid', value='0'},
                                 {label='Forbid', value='1', pen=COLOR_LIGHTRED},
                                 {label='Unforbid', value='2', pen=COLOR_LIGHTBLUE}},
```

Reply:
is `:getOptionValue()` expensive? After changing to your suggestion this seems nice and simple,
![image](https://github.com/DFHack/dfhack/assets/150195674/315b0c80-9f93-4f64-91a2-eaec77bc2aba)

But unless `:getOptionValue()` is expensive then it seems like we'd want to do the same for the other values? Once that was done, would we really even did to getStockpileConfigs (which does hit every watched stockpile when I'd think we just need this one)?

I also couldn't quite figure out how to pass the value and the subviews' `view_id` to the `on_change` callback function. Note that while all my coding abilities are _very very rusty_, this is my first time with lua (and I already love it and hate it! lol). I did see that it will call the callback with (newvalue, oldvalue) but whenever I tried to reference value or view_id I couldn't figure out how.